tmail-backend/tmail-third-party/open-ai/src/main/java/com/linagora/tmail/mailet/ChatLanguageModelFactory.java

```diff
@@ -0,0 +1,28 @@
+package com.linagora.tmail.mailet;
+
+import java.net.URL;
+import java.util.Optional;
+
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.openai.OpenAiChatModel;
+
+public class ChatLanguageModelFactory {
+
+    public ChatLanguageModel createChatLanguageModel(AIBotConfig config) {
+        String apiKey = config.getApiKey();
+        LlmModel llmModel = config.getLlmModel();
+        Optional<URL> baseURLOpt = config.getBaseURL();
+
+        return switch (llmModel.llm()) {
+            case OPEN_AI -> createOpenAILanguageModel(apiKey, llmModel.modelName(), baseURLOpt.map(URL::toString).orElse(null));
```

Comment:
I'm missing something here. Can you tell me what happens in case of having baseURL empty? Where do we get the endpoint used for AI API?

Also I would advocate the usage of null if possible

Reply:
Please see the comment above.